contributing-guides/style-guide.md

```diff
@@ -94,6 +94,12 @@ Use backticks on the following:
 - Extensions, ex. `.dll`.
 - Commands, ex. `ls`.
 
+## Imperative Mood
+
+Example descriptions have to be phrased in imperative mood.  
```

Comment:
No, as a Spanish speaker I think it's `indicative`, or in `es`/`pt` words: `presente indicativo`. This form match the imperative.

Reply:
@navarroaxel there may be some terminology mixup, so perhaps going by example would make it easier to agree on what form we want. This is how I understand the options we've discussed so far (please forgive any grammatical terminology mistakes):

| verb form | Portuguese | Spanish | French | English |
|-----------|------------|---------|--------|---------|
| indicative | Ele apaga o ficheiro. | Él borra el archivo | Il supprime un fichier. | It deletes a file. |
| imperative | Apaga o ficheiro! | Borra el archivo! | Supprime le fichier! | Delete the file! |
| infinitive | Apagar o ficheiro. | Borrar el archivo. | Supprimer le fichier. | (To) Delete the file. |

In the first row, I used the third person as I imagine the idea is for the sentence to describe what the command is doing.

As you mentioned, indeed the present indicative form is identical to the imperative (at least in the Romance languages in the table above); but if we use the indicative, it becomes ambiguous to the person reading the page whether the descriptions are meant as statements or as orders. Besides, the infinitive form does seem more appropriate even if we discount the potential ambiguity of the indicative form.

In English, the infinitive is identical to the current recommendation of using the imperative, so the change I'm proposing would IMO improve the non-English pages, while not affecting the English pages. Let me know what you think! (also @marchersimon and @sbrl.)

Reply:
If I understand your explanation: yes, I prefer recommend present indicate for translations.

Reply:
In German for example, the infinitive form doesn't work here, we have to use the imperative form. So we would have to add a recommendation for each language. This could be done by having the `style-guide` translated like we already started doing in #6344 and #5952, but that's pretty unmaintainable. What do you think of having an own file for language specific rules which could look something like this:

```Markdown
# en

## Sentence structure

The infinitive mood is used, for example `Delete a file` or `Create a new directory`.

## Some other rules

# de

## Sentence structure

The imperative mood is used, for example `Lösche eine Datei` or `Erstelle ein neues Verzeichnis`.

## Some other rules
```

We could also group it in languages so we don't have to repeat the same rules for some languages, but as the file grows, I think it would get hard to navigate to the right section.

Reply:
Good points, @marchersimon. Do you believe there are other rules in the style guide that would benefit from separate instructions per language? Otherwise I would suggest splitting only the "Sentence structure" section into language-specific subsections, rather than splitting the entire document into language-specific sections and duplicating each of the existing sections within each language section.

> If I understand your explanation: yes, I prefer recommend present indicate for translations.

Huh, interesting. For Portuguese, at least, the infinitive works better than the indicative, IMO. I guess we would need to discuss each language's recommendations with the contributors and maintainers that are native speakers of that language so as to reach a consensus.

Reply:
Some things I can think of:

* Chinese punctuation
* Hnidi punctuation
* The space before punctuation in French
* Umlauts in German and other languages

But I'm not sure if we really should make a new file for those. But on the other side, if we would put the sentence structure in the `style-guide` and add an example for each language, it would take a lot of space and make it more unpleasant to read.

Reply:
Oh yeah, I didn't think of those. I withdraw the suggestion to split only the sentence structure section.

> I'm not sure if we really should make a new file for those.

At this point, I would recommend having the contents of the style guide all in one file, and the translations to be just that — direct translations, rather than adaptations to cover language-specific variations. This would make it easier for maintainers to look up rules for specific languages they may have some familiarity with but not full fluency, and would make it viable to ensure the translations remain relatively in sync with the main doc.

That said, it might be a good choice to have all the language-specific recommendations at the end of the style guide, and make references to them whenever it's relevant. So for example, the "Token syntax" section would only contain instructions for English pages, but it would have a note at the end saying something like "see section foobar for language-specific recommendations about the token syntax" (which e.g. for Portuguese, would explain that accented characters are acceptable in token names, etc.).

We could even use the `<details>` element for the language-specific instructions, so that they would become collapsible and wouldn't take up much space in the page. Like this:

> # Style guide
> 
> ## Section 1
> Blah blah blah.
>
> ## Section 2
> Blah blah blah.
>
> ## Language-specific guidelines
>
> <details><summary>Portuguese</summary>
> Portuguese-specific guidelines
> </details>
>
> <details><summary>French</summary>
> French-specific guidelines
> </details>
>
> <details><summary>German</summary>
> German-specific guidelines
> </details>



WDYT?

Reply:
Looks good. But I can't guarantee I can get to it in the net few days.

Reply:
Not up on complex grammar rules myself, but I suggest the imperative mood if possible (as we do currently - e.g. Delete a file`, `Do a thing`?). If the language doesn't have such a concept, then the infinitive mood will do also.